Cargo.toml

```diff
@@ -295,6 +296,74 @@ members = [
 ]
 default-members = ["cmd/tikv-server", "cmd/tikv-ctl"]
 
+[workspace.dependencies]
```

Comment:
Only following needs to be configured with `default-features = false`:
```
> grep 'default =' **/*.toml | grep engine
Cargo.toml:default = ["test-engine-kv-rocksdb", "test-engine-raft-raft-engine", "cloud-aws", "cloud-gcp", "cloud-azure"]
cmd/tikv-ctl/Cargo.toml:default = ["test-engine-kv-rocksdb", "test-engine-raft-raft-engine", "cloud-aws", "cloud-gcp", "cloud-azure"]
cmd/tikv-server/Cargo.toml:default = ["test-engine-kv-rocksdb", "test-engine-raft-raft-engine", "cloud-aws", "cloud-gcp", "cloud-azure"]
components/backup-stream/Cargo.toml:default = ["test-engine-kv-rocksdb", "test-engine-raft-raft-engine"]
components/backup/Cargo.toml:default = ["test-engine-kv-rocksdb", "test-engine-raft-raft-engine", "cloud-aws", "cloud-gcp", "cloud-azure"]
components/cdc/Cargo.toml:default = ["test-engine-kv-rocksdb", "test-engine-raft-raft-engine"]
components/engine_test/Cargo.toml:default = ["test-engine-kv-rocksdb", "test-engine-raft-raft-engine"]
components/engine_traits_tests/Cargo.toml:default = ["test-engine-kv-rocksdb", "test-engine-raft-raft-engine"]
components/raftstore-v2/Cargo.toml:default = ["test-engine-kv-rocksdb", "test-engine-raft-raft-engine"]
components/raftstore/Cargo.toml:default = ["test-engine-kv-rocksdb", "test-engine-raft-raft-engine", "engine_rocks"]
components/test_coprocessor/Cargo.toml:default = ["test-engine-kv-rocksdb", "test-engine-raft-raft-engine"]
components/test_raftstore/Cargo.toml:default = ["test-engine-kv-rocksdb", "test-engine-raft-raft-engine", "cloud-aws", "cloud-gcp", "cloud-azure"]
components/test_storage/Cargo.toml:default = ["test-engine-kv-rocksdb", "test-engine-raft-raft-engine"]
components/tikv_kv/Cargo.toml:default = ["test-engine-kv-rocksdb", "test-engine-raft-raft-engine"]
tests/Cargo.toml:default = ["failpoints", "testexport", "test-engine-kv-rocksdb", "test-engine-raft-raft-engine", "cloud-aws", "cloud-gcp", "cloud-azure"]
```

Reply:
It's really the same discussion again. We don't know which package will introduce new default features. Disabling them all doesn't seem very annoying compared to debugging the feature conflict manually.

Reply:
You have added the new CI pipeline, new mis-configured default features should be detected before being merged.

Reply:
For people aware of the internal reasoning to disable default features, my approach reduces mental load (the writer doesn't need to check if the package indeed has features).
For people unaware of it, my approach saves him from needing to understand how to differentiate the two conditions.
Of course we can always add some comments to explain the reasons behind if you think the "unnecessary" annotations is misrepresenting the fact.

Reply:
Disabling default features is not consistent and causing confusions. We will put more dependencies into workspace dependencies including kvproto, grpcio, etc. Those also have default features, but will not be disabled by default. 

We should keep code clear and understood.